iothub/device/src/Transport/HttpClientHelper.cs

```diff
@@ -434,7 +432,10 @@ private static void InsertEtag(HttpRequestMessage requestMessage, IETagHolder en
             if (!_usingX509ClientCert)
             {
                 string authHeader = await _authenticationHeaderProvider.GetPasswordAsync().ConfigureAwait(false);
-                msg.Headers.Add(HttpRequestHeader.Authorization.ToString(), authHeader);
+                if (!string.IsNullOrWhiteSpace(authHeader))
```

Comment:
In the unlikely event that we do get null back, don't try to add it as a header, because it would be invalid.